### PR TITLE
Improve the HTML formatting

### PR DESCRIPTION
**Pull request by davidjameshumphreys**
Fixes for #111 using Hiccup to generate the HTML
Also improves the code for #184 to sort summary columns on the namespace (default), form percentage and line percentage.

No tests added.

### CODE REVIEW

**Comment by lvh** [comment]
Heya; for some reason I wasn't getting notifications on this repo, sorry for the delay.

Some things I like here, some tweaks I'd like to see. I'll add them as inline comments.

For those following along at home: I compared the shasum of the alleged jquery file and the one from the website. This succeeded for jquery, but not the table sorting plugin. Any idea where I can get the file from upstream? Unfortunately while jquery puts a version in the filename, the table sorter does not :/

File: cloverage/src/cloverage/report/html.clj [hunk 1]
```diff
@@ -3,18 +3,22 @@
...
+(def JQUERY-SORTING "jquery.tablesorter.min.js")
+(def JQUERY "jquery-3.3.1.min.js")
+(def START "start.js")
+(def STYLESHEET "coverage.css")
+(defn css [& args]
+  (->> args
+       (into {})
+       (map (fn [[k v]]
+              (str (name k) \: v)))
+       (cs/join \;)))
```
Comment (lvh) [inline cloverage/src/cloverage/report/html.clj]: I can't follow how this works. You appear to call the function with a map, but this takes `& args`, so now `args` is `({:width 1})` or whatever. Why is that something you can pass to `(into {})`?

File: cloverage/src/cloverage/report/html.clj [hunk 2]
```diff
@@ -3,18 +3,22 @@
...
-   \& "&amp;"
-   \< "&lt;"
-   \> "&gt;"
-   \" "&quot;"
-   \' "&#x27;"
-   \/ "&#x2F;"})
+(def JQUERY-SORTING "jquery.tablesorter.min.js")
+(def JQUERY "jquery-3.3.1.min.js")
+(def START "start.js")
+(def STYLESHEET "coverage.css")
```
Comment (lvh) [inline cloverage/src/cloverage/report/html.clj]: I'd be marginally happier if these weren't randomly uppercased :)

**Comment by davidjameshumphreys** [comment]
> Any idea where I can get the file from upstream?

- JQuery Table Sorter: https://mottie.github.io/tablesorter/docs/
- Zipped Distribution: https://github.com/Mottie/tablesorter/archive/master.zip

The checksum will fail for the table sorter because the lint step in the branch build required a final newline.

I've been thinking about this a bit more: instead of just folding more JS & Hiccup into the project, maybe this could be made as a standalone library?

The PR for #226 would allow for this to slot in without much fuss.

**Comment by lvh** [comment]
That sounds fine to me; happy to take a PR to include a link in the README as well for visibility. Should we close this one then?

**Pull request closed without merging**